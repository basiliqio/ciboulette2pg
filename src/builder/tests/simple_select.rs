use super::*;

#[test]
fn simple() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("cte_main_insert"),
    );
    builder
        .gen_select(&table, vec![("first-name", None), ("last-name", None)])
        .unwrap();
    let res = builder.build().unwrap();

    insta::assert_debug_snapshot!(res);
}

#[test]
fn alias() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("cte_main_insert"),
    );
    builder
        .gen_select(
            &table,
            vec![("first-name", Some("fn")), ("last-name", Some("ln"))],
        )
        .unwrap();
    let res = builder.build().unwrap();

    insta::assert_debug_snapshot!(res);
}

#[test]
fn all() {
    let mut builder = Ciboulette2PostgresBuilder::default();
    let table = Ciboulette2PostgresTableSettings::new(
        Cow::Borrowed("id"),
        Cow::Borrowed("uuid"),
        Some(Cow::Borrowed("public")),
        Cow::Borrowed("cte_main_insert"),
    );
    builder.gen_select(&table, vec![]).unwrap();
    let res = builder.build().unwrap();

    insta::assert_debug_snapshot!(res);
}
