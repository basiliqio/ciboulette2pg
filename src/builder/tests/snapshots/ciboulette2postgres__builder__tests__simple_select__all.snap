---
source: src/builder/tests/simple_select.rs
expression: res

---
(
    "SELECT * FROM \"public\".\"cte_main_insert\"",
    Ciboulette2SqlArguments {
        inner: [],
    },
)
